File: climber-ng/src/app/_Models/wall.ts
import { Color } from "./color";
import { Grade } from "./grade";
import { Rating } from "./rating";
import { User } from "./user";

export class Wall {
    id: string;
    name: string;
    grade: Grade;
    color: Color;
    description?: string;
    created: Date;
    removed?: Date;
    userId: number;
    rating?: Rating;
    anchor: number;
    typesJSON: string;
}

File: climber-api/Models/Wall.cs
using climber_api.Models;
using WebApi.Entities;
using WebApi.Entities.Enums;

namespace WebApi.Models
{
    public class Wall
    {
        public Guid Id { get; set; }
        public Grade Grade { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public Color Color { get; set; }
        public DateTime Created { get; set; }
        public DateTime? Removed { get; set; }
        public int UserId { get; set; }
        public Rating Rating { get; set; }
        public int Anchor { get; set; }
        public string TypesJSON { get; set; } //Compress a Type[] as JSON String to store in DB

    }
}
